### PR TITLE
Invalid emails from GitHub

### PR DESCRIPTION
**Pull request by carols10cents**
This supersedes #1251.

### CODE REVIEW

**Comment by ashleygwilliams** [comment]
I wanted to raise this on the issue but I think we should discuss why we aren’t requiring a valid email to sign up? Would be a good topic for an upcoming meeting

**Comment by carols10cents** [comment]
Ok, in this PR I started from master, added tests [that fail](https://travis-ci.com/rust-lang/crates.io/jobs/176655756#L1036), then pulled in #1251 and updated it to compile atop master with the moving around and 2018-editioning we've done, and now it passes 🎉

**Comment by carols10cents** [comment]
@ashleygwilliams there are things you can do on crates.io like follow a crate that we don't really need an email address for

**Comment by sgrif** [comment]
Thanks for doing the work to update this

**Comment by najamelan** [comment]
How does this work for things where you do require a valid email? Does it still rely on the public profile email from github, because in that case it's pushed the problem further down the pipeline. The good thing is that first thing on the user profile is the possibility to edit the account email address, so if people can sign up, they can then provide an email without relying on the github api, so that looks like a major improvement!

Thanks for taking this on.

**Comment by jtgeibel** [comment]
I've deployed this PR to https://jtgeibel-staging-crates-io.herokuapp.com/ for anyone who would like to test out this change further.

Also, cc #1404 which we should close once this PR is merged.  I'm r+ on this PR, but I think we're looking to deploy current master before merging more changes.

**Comment by Matthias247** [comment]
I (the person with the invalid mail address) tried logging into @jtgeibel's staging area, and can login successfully now. I also get to the dialog to change my mail address. That one doesn't send me a verification mail on my real address, but it might be due to the staging server not sending emails?

Thanks for working on that so quickly!

File: src/email.rs [hunk 1]
```diff
@@ -100,3 +115,24 @@ fn send_email(recipient: &str, subject: &str, body: &str) -> CargoResult<()> {
...
+            "String.Format(\"{0}.{1}@live.com\", FirstName, LastName)",
+            "test",
+            "test",
+        );
+        assert!(result.is_err());
+    }
+
+    #[test]
+    fn sending_to_valid_email_succeeds() {
+        let result = send_email("someone@example.com", "test", "test");
```
[the remark this answers is not in the document]
Comment (sgrif) [reply src/email.rs]: Ah I guess we're just writing to a tempfile. I'm wondering if we could make it a bit more specific that this is only testing the lettre part of our stack, but I guess at the end of the day that really is just the whole thing and we can trust that it's behaving the same in the file sender vs smtp

**Comment by sgrif** [comment]
Sorry, I guess I had started a review rather than actually leaving that comment a few days ago >_>

@bors r+

**Comment by sgrif** [comment]
I'll deploy this once the current deploy has been stable for a bit. Probably won't get to it until tomorrow since it's already late afternoon over here.

**Pull request merged**